packages/jss-plugin-syntax-nested/src/index.js

```diff
@@ -87,7 +88,7 @@ export default function jssNested(): Plugin {
         let selector = replaceParentRefs(prop, styleRule.selector)
         // Lazily create the ref replacer function just once for
         // all nested rules within the sheet.
-        if (!replaceRef) replaceRef = getReplaceRef(container)
+        if (!replaceRef) replaceRef = getReplaceRef(container, sheet)
```

Comment:
My comment before was overwritten here, so we can make sure above that container is either container rule or a sheet, without using a second argument, getRule method is on both.

Reply:
Should we maybe always use the sheet to only ever reference rules from the top level?

Reply:
I'm wondering why we need change onProcessStyle(style, rule) to onProcessStyle(style, rule, sheet?: StyleSheet)?  It seems sheet value is not an input, could get got from the first 2 argument.

Reply:
It is a valid argument. It's the top level sheet which I think we want here.

Reply:
actually I think you want always the sheet, we should only support referencing top level rules

Reply:
looking at this example, this would be unexpected I think, so it should b container or sheet ...